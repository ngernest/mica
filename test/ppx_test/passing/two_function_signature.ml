module type S = sig
  type 'a t

  val f : 'a t -> 'a t
  val g : int -> 'a t
end
