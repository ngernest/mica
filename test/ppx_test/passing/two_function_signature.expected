module type S  = sig type 'a t val f : 'a t -> 'a t val g : int -> 'a t end
[@@deriving mica_types]
include
  struct
    type expr =
      | F of expr 
      | G of int 
    type ty =
      | IntT 
    let gen_expr ty =
      let open Core in
        let open Quickcheck.Generator in
          let open Let_syntax in
            size >>=
              (fun x ->
                 match ty with | IntT -> of_list [F expr__001_; G int__002_])
  end[@@ocaml.doc "@inline"][@@merlin.hide ]
